vllm/pooling_params.py

```diff
@@ -7,7 +7,7 @@ class PoolingParams(
         msgspec.Struct,
         omit_defaults=True,  # type: ignore[call-arg]
         array_like=True):  # type: ignore[call-arg]
-    """Pooling parameters for pooling.
+    """Pooling parameters for embeddings API.
```

Comment:
I might be missing something, but the `additional_data` attribute doesn't seem to be used anywhere. Which is good, because it can by anything and is passed without validation from the request to the `Pooler.forward()` method as part of the `PoolingMetadata` object. If there is no use case for this, can we remove it in this PR?

Reply:
@robertgshaw2-neuralmagic originally added this (https://github.com/vllm-project/vllm/pull/4800#issuecomment-2119278945). I am not sure whether this is still relevant since we can now set the pooling strategy via CLI.

Reply:
@robertgshaw2-neuralmagic can you comment on this?

Reply:
Meanwhile let's merge this PR first.